test/extensions/filters/http/ext_proc/ext_proc_grpc_fuzz.cc

```diff
@@ -0,0 +1,303 @@
+#include "test/integration/http_integration.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace ExternalProcessing {
+
+using envoy::config::core::v3::HeaderValue;
+using envoy::config::core::v3::HeaderValueOption;
+using envoy::extensions::filters::http::ext_proc::v3alpha::ProcessingMode;
+using envoy::service::ext_proc::v3alpha::BodyResponse;
+using envoy::service::ext_proc::v3alpha::CommonResponse;
+using envoy::service::ext_proc::v3alpha::HeadersResponse;
+using envoy::service::ext_proc::v3alpha::HttpBody;
+using envoy::service::ext_proc::v3alpha::HttpHeaders;
+using envoy::service::ext_proc::v3alpha::HttpTrailers;
+using envoy::service::ext_proc::v3alpha::ImmediateResponse;
+using envoy::service::ext_proc::v3alpha::ProcessingRequest;
+using envoy::service::ext_proc::v3alpha::ProcessingResponse;
+using envoy::service::ext_proc::v3alpha::TrailersResponse;
+using envoy::type::v3::StatusCode;
+
+using Http::LowerCaseString;
+
+// The buffer size for the listeners
+static const uint32_t BufferSize = 100000;
+
+// These tests exercise the ext_proc filter through Envoy's integration test
+// environment by configuring an instance of the Envoy server and driving it
+// through the mock network stack.
+
+class ExtProcIntegrationFuzz : public HttpIntegrationTest,
+                               public Grpc::BaseGrpcClientIntegrationParamTest {
+public:
+  ExtProcIntegrationFuzz(Network::Address::IpVersion ip_version, Grpc::ClientType client_type)
+      : HttpIntegrationTest(Http::CodecType::HTTP2, ip_version) {
+    ip_version_ = ip_version;
+    client_type_ = client_type;
+  }
+
+  void TearDown() {
+    cleanupUpstreamAndDownstream();
+    test_processor_.shutdown();
+  }
+
+  Network::Address::IpVersion ipVersion() const override { return ip_version_; }
+  Grpc::ClientType clientType() const override { return client_type_; }
+
+  void initializeFuzzer(bool autonomous_upstream) {
+    autonomous_upstream_ = autonomous_upstream;
+    autonomous_allow_incomplete_streams_ = true;
+    initializeConfig();
+    HttpIntegrationTest::initialize();
+  }
+
+  void initializeConfig() {
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      // Create a cluster for our gRPC server pointing to the address that is running the gRPC
+      // server.
+      auto* processor_cluster = bootstrap.mutable_static_resources()->add_clusters();
+      processor_cluster->set_name("ext_proc_server");
+      processor_cluster->mutable_load_assignment()->set_cluster_name("ext_proc_server");
+      auto* address = processor_cluster->mutable_load_assignment()
+                          ->add_endpoints()
+                          ->add_lb_endpoints()
+                          ->mutable_endpoint()
+                          ->mutable_address()
+                          ->mutable_socket_address();
+      address->set_address("127.0.0.1");
```

Comment:
This is hardcoding IPv4, but Envoy integration tests are parameterized by IP version. Look at `Network::Test::getLoopbackAddressString` and other example in integration tests of how to make this IPv6 safe.